### PR TITLE
Fix the plugin references to work with packages

### LINKED ISSUE #15811: Plugins should include ExcludeAssets=runtime for project references to shared assemblies
The guide along with the sample it uses has this for non-private project references:
``` XML
<ProjectReference Include="PluginBase">
  <Private>false</Private>
</ProjectReference>
```

This works for the sample itself, but it doesn't always correctly translate for more complex projects. Specifically if `PluginBase` would have a `PackageReference`, the "Private=false" doesn't correctly propagate over to the `PackageReference` and the output contains "traces" of the NuGet package (in the `.deps.json`). This can cause issues when loading plugins.

The workaround seems to be to include "ExcludeAssets=runtime" as well, so the project reference would look like this:
``` XML
<ProjectReference Include="PluginBase">
  <Private>false</Private>
  <ExcludeAssets>runtime</ExcludeAssets>
</ProjectReference>
```

The exact workaround needs to be validated with the product team (@nguerrera).
Even though this is not necessary for the sample in question, fixing it here will make people using the guide more likely to a working state at first try.

We're still trying to figure out if this is an actual product bug or if this is a desired behavior. Either way for .NET Core 3.0 this is how it works, and so fixing the guide is preferable.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 06023407-704f-0ace-e89a-eedd9f95a007
* Version Independent ID: f25a6e1a-cb10-867b-0522-a377bf9d016b
* Content: [Create a .NET Core application with plugins - .NET Core](https://docs.microsoft.com/en-us/dotnet/core/tutorials/creating-app-with-plugin-support)
* Content Source: [docs/core/tutorials/creating-app-with-plugin-support.md](https://github.com/dotnet/docs/blob/master/docs/core/tutorials/creating-app-with-plugin-support.md)
* Product: **dotnet-core**
* GitHub Login: @jkoritzinsky
* Microsoft Alias: **jekoritz**

### PR DESCRIPTION
**Pull request by vitek-karas**
## Summary

For plugins to correctly reference other projects they need to specify both Private=false and ExcludeAssets=runtime so that it works for both project and package references.

Fixes #15811

### CODE REVIEW

File: docs/core/tutorials/creating-app-with-plugin-support.md [hunk 1]
```diff
@@ -245,14 +245,17 @@ In between the `<Project>` tags, add the following elements:
...
 </ItemGroup>
 ```
 
 The `<Private>false</Private>` element is important. This tells MSBuild to not copy *PluginBase.dll* to the output directory for HelloPlugin. If the *PluginBase.dll* assembly is present in the output directory, `PluginLoadContext` will find the assembly there and load it when it loads the *HelloPlugin.dll* assembly. At this point, the `HelloPlugin.HelloCommand` type will implement the `ICommand` interface from the *PluginBase.dll* in the output directory of the `HelloPlugin` project, not the `ICommand` interface that is loaded into the default load context. Since the runtime sees these two types as different types from different assemblies, the `AppWithPlugin.Program.CreateCommands` method won't find the commands. As a result, the `<Private>false</Private>` metadata is required for the reference to the assembly containing the plugin interfaces.
 
+Similarly the `<ExcludeAssets>runtime</ExcludeAssets>` element is also important if the `PluginBase` references other packages. This setting has the same effect as `<Private>false</Private>` but works on package references in the transitive closure of dependencies of the `PluginBase` project.
```
Comment (mairaw) [inline docs/core/tutorials/creating-app-with-plugin-support.md]:
```suggestion
Similarly, the `<ExcludeAssets>runtime</ExcludeAssets>` element is also important if the `PluginBase` references other packages. This setting has the same effect as `<Private>false</Private>` but works on package references in the transitive closure of dependencies of the `PluginBase` project.
```

File: docs/core/tutorials/creating-app-with-plugin-support.md [hunk 2]
```diff
@@ -245,14 +245,17 @@ In between the `<Project>` tags, add the following elements:
...
+
 Now that the `HelloPlugin` project is complete, we should update the `AppWithPlugin` project to know where the `HelloPlugin` plugin can be found. After the `// Paths to plugins to load` comment, add `@"HelloPlugin\bin\Debug\netcoreapp3.0\HelloPlugin.dll"` as an element of the `pluginPaths` array.
```
Comment (mairaw) [inline docs/core/tutorials/creating-app-with-plugin-support.md]: Unrelated to your change, but our content typically uses you as we're talking to the reader and not we.

```suggestion
Now that the `HelloPlugin` project is complete, you should update the `AppWithPlugin` project to know where the `HelloPlugin` plugin can be found. After the `// Paths to plugins to load` comment, add `@"HelloPlugin\bin\Debug\netcoreapp3.0\HelloPlugin.dll"` as an element of the `pluginPaths` array.
```

**Review by jkoritzinsky — APPROVED** [review]

**Review by elinor-fung — APPROVED** [review]

File: docs/core/tutorials/creating-app-with-plugin-support.md [hunk 3]
```diff
@@ -245,15 +245,18 @@ In between the `<Project>` tags, add the following elements:
...
 
-Now that the `HelloPlugin` project is complete, we should update the `AppWithPlugin` project to know where the `HelloPlugin` plugin can be found. After the `// Paths to plugins to load` comment, add `@"HelloPlugin\bin\Debug\netcoreapp3.0\HelloPlugin.dll"` as an element of the `pluginPaths` array.
+Similarly, the `<ExcludeAssets>runtime</ExcludeAssets>` element is also important if the `PluginBase` references other packages. This setting has the same effect as `<Private>false</Private>` but works on package references in the transitive closure of dependencies of the `PluginBase` project.
```
Comment (tdykstra) [inline docs/core/tutorials/creating-app-with-plugin-support.md]: Can we expect the intended audience of this article to be familiar with the phrase "transitive closure of dependencies"? If not, a brief explanation might be helpful.

**Review by tdykstra — COMMENTED** [review]
LGTM except for one question.

File: docs/core/tutorials/creating-app-with-plugin-support.md (hunk 3 above)
Comment (vitek-karas) [reply docs/core/tutorials/creating-app-with-plugin-support.md]: Good point - I changed the wording - hopefully it's easier to understand now.

File: docs/core/tutorials/creating-app-with-plugin-support.md [hunk 4]
```diff
@@ -245,15 +245,18 @@ In between the `<Project>` tags, add the following elements:
...
 
-Now that the `HelloPlugin` project is complete, we should update the `AppWithPlugin` project to know where the `HelloPlugin` plugin can be found. After the `// Paths to plugins to load` comment, add `@"HelloPlugin\bin\Debug\netcoreapp3.0\HelloPlugin.dll"` as an element of the `pluginPaths` array.
+Similarly, the `<ExcludeAssets>runtime</ExcludeAssets>` element is also important if the `PluginBase` references other packages. This setting has the same effect as `<Private>false</Private>` but works on package references which the `PluginBase` project or one of its dependencies may include.
```
Comment (tdykstra) [inline docs/core/tutorials/creating-app-with-plugin-support.md]:
```suggestion
Similarly, the `<ExcludeAssets>runtime</ExcludeAssets>` element is also important if the `PluginBase` references other packages. This setting has the same effect as `<Private>false</Private>` but works on package references that the `PluginBase` project or one of its dependencies may include.
```

**Review by tdykstra — APPROVED** [review]
Looks good.

**Pull request merged**